lint-tasks.yml

```
@@ -1,127 +1,134 @@
+          FLAGS: ["--dry-run"]
+          INCLUDE_FILENAME_PATTERNS: ["*.cpp", "*.h", "*.hpp", "*.inc"]
+          ROOT_PATHS: *cpp_format_source_files
+          VENV_DIR: "{{.G_LINT_VENV_DIR}}"
 
-  cpp-format-fix:
+  fix-cpp-format:
     sources: *cpp_format_source_files
-    deps: ["cpp-configs", "venv"]
+    deps: ["cpp-lint-configs", "venv"]
     cmds:
-      - task: "clang-format"
+      - task: ":utils:cpp-lint:clang-format"
         vars:
-          FLAGS: "-i"
-
-  cpp-static-check:
-    # Alias task to `cpp-static-fix` since we don't currently support automatic fixes.
-    # NOTE: clang-tidy does have the ability to fix some errors, but the fixes can be inaccurate.
-    # When we eventually determine which errors can be safely fixed, we'll allow clang-tidy to
-    # fix them.
-    aliases: ["cpp-static-fix"]
-    sources:
-      # Dependency sources
-      - "{{.G_CLP_FFI_JS_BUILD_DIR}}/_deps/*-src/**/*"
-      - "{{.G_SRC_DIR}}/submodules/**/*"
-
-      # clp-ffi-js sources
-      - "{{.G_SRC_CLP_FFI_JS_DIR}}/**/*.cpp"
-      - "{{.G_SRC_CLP_FFI_JS_DIR}}/**/*.h"
-      - "{{.G_SRC_CLP_FFI_JS_DIR}}/**/*.hpp"
-
-      # Other sources
-      - "{{.G_CLP_FFI_JS_BUILD_DIR}}/compile_commands.json"
-      - "{{.ROOT_DIR}}/Taskfile.yml"
-      - "{{.TASKFILE}}"
-      - "tools/yscope-dev-utils/lint-configs/.clang-tidy"
-    deps: [":config-cmake-project", "cpp-configs", "venv"]
+          FLAGS: ["-i"]
+          INCLUDE_FILENAME_PATTERNS: ["*.cpp", "*.h", "*.hpp", "*.inc"]
+          ROOT_PATHS: *cpp_format_source_files
+          VENV_DIR: "{{.G_LINT_VENV_DIR}}"
+
+  check-cpp-static-diff:
+    # NOTE: We alias the fix task until we determine which clang-tidy fixes can be applied safely.
+    aliases: ["fix-cpp-static-diff"]
+    deps:
+      - ":config-cmake-project"
+      - "cpp-lint-configs"
+      - "venv"
     cmds:
-      - task: "clang-tidy"
-
-  yml:
+      - task: ":utils:cpp-lint:clang-tidy-diff"
+        vars:
+          FLAGS: &clang_tidy_flags
+            - "--config-file '{{.ROOT_DIR}}/.clang-tidy'"
+            - "-p '{{.G_CLP_FFI_JS_BUILD_DIR}}/compile_commands.json'"
+            - >-
+              $(cat "{{.G_CLP_FFI_JS_BUILD_DIR}}/compiler-extra-args.txt" \
+                | tr ' ' '\n' \
+                | sed 's/^/--extra-arg /' \
+                | tr '\n' ' ')
+          OUTPUT_DIR: "{{.G_LINT_CLANG_TIDY_DIR}}"
+          VENV_DIR: "{{.G_LINT_VENV_DIR}}"
+
+  check-cpp-static-full:
+    # NOTE: We alias the fix task until we determine which clang-tidy fixes can be applied safely.
+    aliases: ["fix-cpp-static-full"]
+    deps:
+      - ":config-cmake-project"
+      - "cpp-lint-configs"
+      - "venv"
+    cmds:
+      - task: ":utils:cpp-lint:clang-tidy-find"
+        vars:
+          FLAGS: *clang_tidy_flags
+          INCLUDE_PATTERNS:
+            - "{{.G_SRC_CLP_FFI_JS_DIR}}/**"
+          OUTPUT_DIR: "{{.G_LINT_CLANG_TIDY_DIR}}"
+          ROOT_PATHS:
+            - "{{.G_SRC_CLP_FFI_JS_DIR}}/**/*"
```

Sorry I should have spotted this in the first review. This suggestion _should_ work, but please double check just in case.

```suggestion
          OUTPUT_DIR: "{{.G_LINT_CLANG_TIDY_DIR}}"
          ROOT_PATHS:
            - "{{.G_SRC_CLP_FFI_JS_DIR}}"
```

relevant doc-string snippets:
> * ROOT_PATHS Paths recursively searched for files to run on.
> * [INCLUDE_PATTERNS] Wildcard patterns where any matching file paths (relative
   to `ROOT_PATHS`) will be included. By default, all paths match.
   
explanations:
* Adding child paths of `G_SRC_CLP_FFI_JS_DIR` to `ROOT_PATHS` is unnecessary because `find` recursively searches each path in `ROOT_PATHS`.
* Having `INCLUDE_PATTERNS` match anything under `G_SRC_CLP_FFI_JS_DIR` is unnecessary as by default `INCLUDE_PATTERNS` will match everything under the `ROOT_PATHS`.